types/pg-cursor/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for pg-cursor 2.5
+// Project: https://github.com/brianc/node-postgres#readme
+// Definitions by: Tiogshi Laj <https://github.com/Tiogshi>, Edited by Elias Fauser <https://github.com/elias-fauser>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
+
+import { Connection, CustomTypesConfig, QueryResult } from "pg";
+
+declare namespace Cursor {
+    interface CursorQueryConfig {
+        /**
+         * By default rows come out as a Record<string,any>.
+         * Pass the string 'array' here to receive each row as an array of values.
+         */
+        rowMode?: "array";
+        /**
+         * Custom type parsers for just this query result.
+         */
+        types?: CustomTypesConfig;
+    }
+    type ResultCallback<RowType> = (err: Error | undefined, rows: RowType[], result: QueryResult) => void;
+}
+
+declare class Cursor<Row = any> {
+    constructor(query: string, values?: any[], config?: Cursor.CursorQueryConfig);
+    submit: (connection: Connection) => void;
+    read: (maxRows: number, callback: Cursor.ResultCallback<Row>) => void;
+    close: (callback: (err: Error) => void) => void;
```

Comment:
```suggestion
    submit(connection: Connection): void;
    read(maxRows: number, callback: Cursor.ResultCallback<Row>): void;
    read(maxRows: number): Promise<Error | RowType[]>;
    close(callback: (err: Error) => void): void;
    close(): Promise<Error>;
```

Reply:
Methods returns promise if callback is not defined
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L208
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L231

Reply:
Thanks for noticing. This was indeed wrong. Added function overloads for the different cases